ros2doctor/ros2doctor/api/__init__.py

```diff
@@ -15,19 +15,68 @@
 from pkg_resources import iter_entry_points
 
 
+class DoctorCheck:
+    """
+    Abstract based class of ros2doctor check modules.
+
+    category method returns a string that includes the module name.
+    check method returns a boolean value based on check result.
+    """
+
+    def category(self):
+        raise NotImplementedError
+
+    def check(self):
+        raise NotImplementedError
+
+
+class DoctorReport:
+    """
+    Abstract based class of ros2doctor report modules.
+
+    category method returns a string that includes the module name.
+    report method returns a Report instance that contains report content.
+    """
+
+    def category(self):
+        raise NotImplementedError
+
+    def report(self):
+        raise NotImplementedError
+
+
+class Report:
+    """Contains report name and content."""
+
+    def __init__(self, name):
+        self.name = name
+        self.items = []
```

Comment:
Recommend using `__slots__` since the number and names of the attributes are fixed.